### PR TITLE
CB-21119 Ranger RAZ - get metrics api via knox proxy

### PR DESCRIPTION
**Pull request by adarshsharma1983**
Expose Ranger Raz via knox proxy

See detailed description in the commit message.

### CODE REVIEW

File: core/src/main/java/com/sequenceiq/cloudbreak/core/bootstrap/service/host/ClusterHostServiceRunner.java [hunk 1]
```diff
@@ -1002,4 +1014,38 @@ public void createCronForUserHomeCreation(StackDto stackDto, Set<String> candida
...
+                if (!datalakes.isEmpty()) {
+                    razEnabled = datalakes.get(0).getRangerRazEnabled();
+                }
+            }
+
+            if (razEnabled) {
+                ExposedService rangerRazService = exposedServiceCollector.getRangerRazService();
+
+                // Find master nodes since Ranger Raz is only going to be installed on master nodes
+                List<String> rangerRazLocations = stackDto.getAliveInstancesInInstanceGroup("master").stream()
```
Comment (doktoric) [inline core/src/main/java/com/sequenceiq/cloudbreak/core/bootstrap/service/host/ClusterHostServiceRunner.java:1036]: I think you should probably select the GATEWAY node right ?

File: core/src/main/java/com/sequenceiq/cloudbreak/core/bootstrap/service/host/ClusterHostServiceRunner.java [hunk 2]
```diff
@@ -1002,4 +1014,38 @@ public void createCronForUserHomeCreation(StackDto stackDto, Set<String> candida
             throw new CloudbreakException(message, e);
         }
     }
+
+    private void addRangerRazServiceIfAvailable(StackDto stackDto, String cmVersion, Map<String, List<String>> serviceLocations, GatewayConfig gatewayConfig) {
+        if (stackDto.getCluster() != null && stackDto.getStack() != null
```
Comment (doktoric) [inline core/src/main/java/com/sequenceiq/cloudbreak/core/bootstrap/service/host/ClusterHostServiceRunner.java:1019]: I think logic can come from the RangerRazDatahubConfigProvider? Probably with some refactor we can easily use the same logic here. (no dup needed here)

File: core/src/main/java/com/sequenceiq/cloudbreak/core/bootstrap/service/host/ClusterHostServiceRunner.java (hunk 1 above)
Comment (adarshsharma1983) [reply core/src/main/java/com/sequenceiq/cloudbreak/core/bootstrap/service/host/ClusterHostServiceRunner.java]: Done

File: core/src/main/java/com/sequenceiq/cloudbreak/core/bootstrap/service/host/ClusterHostServiceRunner.java (hunk 2 above)
Comment (adarshsharma1983) [reply core/src/main/java/com/sequenceiq/cloudbreak/core/bootstrap/service/host/ClusterHostServiceRunner.java]: Done

**Review by doktoric — APPROVED** [review]

**Pull request merged**